modules/channel-messenger/src/backend/messenger.ts

```diff
@@ -0,0 +1,253 @@
+import axios, { AxiosInstance } from 'axios'
+import bodyParser from 'body-parser'
+import * as sdk from 'botpress/sdk'
+import crypto from 'crypto'
+import { Router } from 'express'
+import _ from 'lodash'
+
+import { Config } from '../config'
+
+const outgoingTypes = ['text', 'typing', 'login_prompt', 'carousel']
+type MessengerAction = 'typing_on' | 'typing_off' | 'mark_seen'
+
+export class MessengerService {
+  private readonly http = axios.create({ baseURL: 'https://graph.facebook.com/v2.6/me' })
+
+  private messengerClients: { [key: string]: MessengerClient } = {}
+  private router: Router
+
+  constructor(private bp: typeof sdk) {}
+
+  initialize() {
+    this.router = this.bp.http.createRouterForBot('channel-messenger', { checkAuthentication: false })
+    this.router.use(
+      bodyParser.json({
+        verify: this._verifySignature.bind(this)
+      })
+    )
+
+    this.router.get('/webhook', this._setupWebhook.bind(this))
```

Comment:
Why use bind instead of ES6 arrows?

Reply:
because these method are part of a class and I don't like having arrow and non-arrow functions side-by-side. Just a matter of preference.